### PR TITLE
#US-1533: PHP 8.1 Upgrade

### PR DESCRIPTION
**Pull request by martinstadelmann**
## Tasks

- [x] Run rector and code beautifying
- [x] Upgrade to `iq_group` `v3.x` releases

### CODE REVIEW

File: iq_whitepaper.module [hunk 1]
```diff
@@ -243,36 +260,35 @@ function iq_whitepaper_views_pre_render(\Drupal\views\ViewExecutable $view) {
...
-
         /** @var \Drupal\group\Entity\GroupInterface[] $groups */
         $groups = [];
         foreach ($group_contents as $group_content) {
           /** @var \Drupal\group\Entity\GroupContentInterface $group_content */
           $group = $group_content->getGroup();
           $groups[$group->id()] = $group;
         }
-        $account = \Drupal\user\Entity\User::load(\Drupal::currentUser()->id());
+        $account = User::load(\Drupal::currentUser()->id());
```
Comment (youpixxl) [inline iq_whitepaper.module:291]: Using User::load or Node:load is generally not recommanded, use \Drupal::entityTypeManager()->getStorage('user')->load instead

File: iq_whitepaper.module [hunk 2]
```diff
@@ -426,7 +445,7 @@ function iq_whitepaper_form_iq_group_user_edit_form_alter(&$form, \Drupal\Core\F
 
   if ($user->hasField('field_iq_group_branches') && !isset($form['branches'])) {
     $vid = 'branches';
-    $terms =\Drupal::entityTypeManager()->getStorage('taxonomy_term')->loadTree($vid);
+    $terms = \Drupal::entityTypeManager()->getStorage('taxonomy_term')->loadTree($vid);
```
Comment (youpixxl) [inline iq_whitepaper.module:448]: GOt I hate that one, but would be fine if you don't have thousand of branches, which will never happen, so fine :-)

File: src/Event/IqWhitepaperEvent.php [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 namespace Drupal\iq_whitepaper\Event;
 
-use Symfony\Component\EventDispatcher\Event;
+use Symfony\Contracts\EventDispatcher\Event;
```
Comment (youpixxl) [inline src/Event/IqWhitepaperEvent.php:5]: Good to know, we probably should update that on all projects. Did you find that out or was it done automatically?

File: src/Form/WhitepaperForm.php [hunk 4]
```diff
@@ -51,7 +50,7 @@ public function buildForm(array $form, FormStateInterface $form_state) {
           'spellcheck' => 'false',
         ],
       ];
-      $termsAndConditions = \Drupal::config('iq_group.settings')->get('terms_and_conditions') ? \Drupal::config('iq_group.settings')->get('terms_and_conditions') : "https://www.sqs.ch/de/datenschutzbestimmungen";
+      $termsAndConditions = \Drupal::config('iq_group.settings')->get('terms_and_conditions') ?: "https://www.sqs.ch/de/datenschutzbestimmungen";
```
Comment (youpixxl) [inline src/Form/WhitepaperForm.php]: We also had this hard coded link to sqs in the iq_group module. As a result, the link on flury redirected to sqs, not good :-) coukd you please just set it null or front?

**Review by youpixxl — CHANGES_REQUESTED** [review]
Hey! looks good to me, just a small change regarding hard coded link :-)

File: src/Event/IqWhitepaperEvent.php (hunk 3 above)
Comment (martinstadelmann) [reply src/Event/IqWhitepaperEvent.php:5]: Rector does this for us auto-magically :)

**Pull request merged**